src/components/Pagination.vue

```diff
@@ -21,9 +34,53 @@
 
         computed: {
             pages() {
-                return this.pagination.totalPages === undefined
-                    ? []
-                    : range(1, this.pagination.totalPages + 1);
+                if (this.pagination.totalPages === undefined) {
+                    return [];
+                }
+
+                const pages = [];
+                let preDots = false;
+                let postDots = false;
+
+                for (let i = 1; i <= this.pagination.totalPages; i++) {
+                    if (this.pagination.totalPages <= 10) {
```

Comment:
We're not big fans of `else` statements.

Could you refactor the loop body to it's own dedicated function so early returns can be used?

Reply:
I've created a dedicated function to generate pagination links.

Let me know if i missed something.

Reply:
I made some more improvements to the pagination. Please take a look and let me know if i missed something.